File: ifgi/scene/HitRecord.py
#!/usr/bin/env python
#
# Copyright 2011 (C) Yamauchi, Hitoshi
#
# HitRecord
#

"""IFGI HitRecord
\file
\brief a hit record
"""

import sys
from ifgi.base import OrthonomalBasis


# HitRecord
class HitRecord(object):
    """hit record. members are public.
    """

    # default constructor
    def __init__(self):
        """default constructor.
        """
        self.dist = sys.float_info.max
        self.intersect_pos = None
        self.hit_primitive = None
        # w component is normal direction of hit point
        self.hit_basis     = None

    # class name
    def get_classname(self):
        """get class name
        \return class name
        """
        return 'HitRecord'


    # string representation
    def __str__(self):
        return 'HitRecord: dist: '+ str(self.dist) +\
            ', pos: ' + str(self.intersect_pos) +\
            ', basis: ' + str(self.hit_basis)



#
# main test
#
# if __name__ == '__main__':
#     pass

File: ifgi/cpp/scene/HitRecord.cpp
//----------------------------------------------------------------------
// ifgi c++ implementation: HitRecord.hh
// Copyright (C) 2010-2012 Yamauchi, Hitoshi
//----------------------------------------------------------------------
/// \file
/// \brief a hit record

#include "HitRecord.hh"

#include <sstream>

namespace ifgi
{
//----------------------------------------------------------------------
// get string representation
std::string HitRecord::to_string() const
{
    std::stringstream sstr;
    sstr << "HitRecord: dist: " << m_dist
         << ", pos: "           << m_intersect_pos
         << ", basis: "         << m_hit_onb.to_string()
         << ", matidx: "        << m_hit_material_index;
    return sstr.str();
}

//----------------------------------------------------------------------
} // namespace ifgi

